modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -528,6 +530,111 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
     }
 #endif  // HAVE_INF_ENGINE
 
+#ifdef HAVE_INF_ENGINE
+    virtual Ptr<BackendNode> initNgraph(const std::vector<Ptr<BackendWrapper> > &inputs,
+                                        const std::vector<Ptr<BackendNode> >& nodes) CV_OVERRIDE
+    {
+        CV_Assert_N(inputs.size() == 1, nodes.size() == 1);
+        auto& ieInpNode = nodes[0].dynamicCast<InfEngineNgraphNode>()->node;
+        std::vector<size_t> dims = ieInpNode->get_shape();
+        CV_Assert(dims.size() == 4 || dims.size() == 5);
+        const int inpCn = dims[1];
+        const int outCn = blobs[0].size[0];
+        const int inpGroupCn = blobs[0].size[1];
+        const int group = inpCn / inpGroupCn;
+
+        auto precision = (preferableTarget == DNN_TARGET_OPENCL_FP16 || preferableTarget == DNN_TARGET_MYRIAD) ?
+                          ngraph::element::f16 : ngraph::element::f32;
+
+        std::vector<size_t> kernel_shape(&blobs[0].size[0], &blobs[0].size[0] + blobs[0].dims);
+        Mat halfs(1, blobs[0].total(), CV_16SC1);
+        if (precision == ngraph::element::f16) {
+            convertFp16(blobs[0], halfs);
+        }
+
+        auto ieWeights = std::make_shared<ngraph::op::Constant>(precision, kernel_shape,
+                         precision == ngraph::element::f16 ? halfs.data : blobs[0].data);
```

Comment:
Maybe we can create some method which can wrap OpenCV's Mat to nGraph's Constant op? Like `wrapToInfEngineBlob`. It will convert to FP16 is necessary.